### ISSUE TITLE
Not able to broadcast video

### ISSUE THREAD

**Issue by somnathbanik** [mmaitre314/IpCamera#1]
Hi,
I have download the code  and deployed on my Laptop (windows 8.1) and Windows Phone 8.1.
The video preview is displaying but no broadcasting of video happening between devices.
If I put something on the Bottom-Right cornet text box it says "Failed to initiate MJPEG camera playback".
Am I missing something.
Please help me.

**Comment by mmaitre314** [comment]
You are using IpCameraTestApp, correct? Some ideas:
- Are the Phone and Laptop connected to the same WiFi network?
- Did you use the IP address + port displayed on the Phone and put it in the text box on the Laptop? I think it should look something like http://x.x.x.x:31415/
- You can also look at the exception raised by the call to HttpMjpegCaptureSource.CreateFromUriAsync() to get a better idea of what happened.

**Comment by somnathbanik** [comment]
Hi,
Thanks for your reply. I have followed your steps and able to stream from 
1. Phone to Phone
2. Laptop to Phone
3. But not Phone to Laptop; getting exception in 
"var client = await HttpMjpegCaptureSource.CreateFromUriAsync(Source.Text);"
as "Exception from HRESULT: 0x80072EFD".

Please put some light on this.

Another thing is, can I use this came app for streaming video from real IP camera (not mobile phone) just change the IP address and port of the camera?

**Comment by mmaitre314** [comment]
Yes, streaming video from real IP cameras should just be a matter of changing the URL passed to HttpMjpegCaptureSource (as long as they are MJPEG IP cameras and not H.264):

``` C#
var client = await HttpMjpegCaptureSource.CreateFromUriAsync("http://localhost:31415/");
```

The HRESULT is ERROR_INTERNET_CANNOT_CONNECT. Most likely either the URL is invalid or there is something on the network getting in the way (firewall, etc.). To narrow down the issue maybe create a test app where HttpClient connects to StreamSocketListener. If that works there is likely an issue in either CameraServer or HttpMjpegCaptureSource.

**Comment by somnathbanik** [comment]
Thanks for your guidance.
Well in case I want to connect a H.264 IP Camera, what change do I need to make in this code. Does this code support H.264 IP Camera as well, if not then is there any other library like this which suppot H.264 IP Camera.

**Comment by mmaitre314** [comment]
I believe H.264 cameras use RSTP for transport instead of HTTP, so adding support for them here won't be trivial. The OS supports H.264 and RTSP, so maybe just set the IP camera URL on MediaElement and try playing? There must be libraries out there handling that otherwise, but I am not aware of them.

Sent from my Windows Phone
